tasks/test-project/tasks.js

```diff
@@ -280,19 +320,27 @@ async function apiTasks(outputPath, { verbose, linkWithLatestFwBuild }) {
     // update directive in contacts.sdl.ts
     const pathPostsSdl = `${OUTPUT_PATH}/api/src/graphql/posts.sdl.ts`
     const contentPostsSdl = fs.readFileSync(pathPostsSdl, 'utf-8')
-    const resultsPostsSdl = contentPostsSdl.replace(
-      /posts: \[Post!\]! @requireAuth([^}]*)@requireAuth/,
-      `posts: [Post!]! @skipAuth
+    const resultsPostsSdl = contentPostsSdl
+      .replace(
+        /posts: \[Post!\]! @requireAuth([^}]*)@requireAuth/,
+        `posts: [Post!]! @skipAuth
       post(id: Int!): Post @skipAuth`
-    )
+      ) // make posts accessible to all
+      .replace(
+        'createPost(input: CreatePostInput!): Post! @requireAuth',
+        'createPost(input: CreatePostInput!): Post! @requireAuth(roles:["ADMIN"])'
+      ) // make creating posts admin only
```

Comment:
@dac09 Because this fixture is also used as a project for local dev, doesn't this effectively block some functionality for the local dev use case? Specifically, there's no way from the UI or otherwise to create a new user and set a role. (I saw that you added the script in the Playwright fixture.)

Could `@requireAuth(roles:["ADMIN"])` be added in the Playwright fixture instead of here? Or is there another option to still have basic functionality flow working at init without people having to dig in and modify things? E.g. how can we keep the "click login -> click signup -> create user -> edit posts" flow at the time a Test-project is created for local dev?

Reply:
My thinkign around this DP, is that modifying code in the e2e tests is a bit of an antipattern, because in the future its going to stop us from running tests in parallel, which is why I prefer to modify the code before launching the tests.

Is it important to have the ability to add posts out of the box? If really needed, its a simple code change in the posts.sdl right?

Reply:
I do believe it's important the Test-project be functional out of the box as that was the initial use-case is still a priority. I added dbAuth primarily for this reason. So any code modifications shouldn't get in the way of flows and using features.

> is that modifying code in the e2e tests is a bit of an antipattern, because in the future its going to stop us from running tests in parallel

I completely understand we want to build a better foundation and be able to eventually run parallel, but it depends, right? E.g. there's already some of that happening here (I realize I've been using "fixture" and not spec): https://github.com/redwoodjs/redwood/pull/4673/files#diff-af563928daa1ee54229ea46b919745293deb7d765faa7cd71cf615be81a96329

And it will be great to be able to run these tests in parallel in the future. But we don't need to run them in parallel now.

To possibilities I see:
1. move the directive with Admin role as a line change code mod in ``tasks/smoke-test/tests/rbacChecks.spec.ts`
2. move the RBAC test to something outside the current functionality, e.g. update the directive for the contacts Query in `contacts.sdl.ts` to be Admin and then create a contact instead of a post

Other options?

Reply:
Let me think about it some more - we've made this mistake of doing what's "okay for now", and it always comes back to bite us! 

I'm also only modifying the project, but not the src in the spec - the reason I think it's an anti pattern when you modify src files (or do something destructive like reset db) - is that it won't just restart the dev server for your test but for all the other tests too. This can cause tests to fail unexpectedly and is super hard to debug - effectively taking the rug out from under the test runner. 

Modifying contacts is certainly an option (but a lot of work) - but let me think about it, maybe there's another way!

Reply:
Completely understood. You've already gone well beyond what I imagined our initial Auth functional tests to be — maybe RBAC functional testing isn't a fit for this context?

> Modifying contacts is certainly an option (but a lot of work)

Yeah, I was only suggesting it as an option if it could be as simple as:
1. apply the Role directive to the SDL query
2. add a contact via the form (instead of adding a post)
3. query contacts via URL — should be denied
4. update user as Admin (run the script)
5. query contacts via URL — should return

The functionality is already in place unless I'm missing something obvious

Reply:
Ok well... I did the hard thing - using contacts to check for RBAC, because I didn't want to deal with fixing the issue when we hit it in the future.

Just incase you're reviewing another PR later, I think we should enforce that we don't modify src files in fixtures or in the test from now on! It makes it harder, but it'll save us _soooo much_ effort in the future!